src/Microsoft.SymbolManifestGenerator/SymbolManifestGenerator.cs

```diff
@@ -0,0 +1,121 @@
+                if (!generator.IsValid())
+                {
+                    tracer.Information($"Could not generate a valid FileKeyGenerator from file '{file.FullName}'. Skipping.");
+                    continue;
+                }
+
+                foreach (SymbolStoreKey clrKey in generator.GetKeys(KeyTypeFlags.ClrKeys))
+                {
+                    string runtimeModuleDirectory = file.DirectoryName;
+                    FileInfo specialFile = GetSymbolFileToAddAdditionalDebugEntry(runtimeModuleDirectory, clrKey);
+                    if (specialFile == null)
+                    {
+                        tracer.Information($"Known special file '{clrKey.FullPathName}' for runtime module '{file.FullName}' does not exist in directory '{runtimeModuleDirectory}'. Skipping.");
+                        continue;
+                    }
+
+                    string basedirRelativePath = Path.Combine(runtimeModuleDirectory.Replace(dir.FullName, string.Empty).TrimStart(Path.DirectorySeparatorChar), specialFile.Name);
+                    string fileHash = CalculateSHA512(specialFile);
+
+                    ManifestDataEntry manifestDataEntry = new()
+                    {
+                        basedirRelativePath = basedirRelativePath,
+                        symbolKey = clrKey.Index,
+                        sha512 = fileHash,
+                        debugInformationLevel = Private
+                    };
+
+                    manifestData.entries.Add(manifestDataEntry);
+                }
+            }
+
+            string manifestDataContent = JsonSerializer.Serialize(manifestData);
+            File.WriteAllText(manifestFileName, manifestDataContent);
+        }
+
+        private static FileInfo GetSymbolFileToAddAdditionalDebugEntry(string runtimeModuleDir, SymbolStoreKey clrKey)
+        {
+            string keyFileNameToMatch = clrKey.FullPathName;
+
+            string fileOnDisk = Directory.EnumerateFiles(runtimeModuleDir)
+                                      .FirstOrDefault(f => Path.GetFileName(f).Equals(keyFileNameToMatch, StringComparison.OrdinalIgnoreCase));
+
+            if (fileOnDisk == null)
+            {
+                return null;
+            }
+
+            return new FileInfo(fileOnDisk);
+        }
+
+        private static string CalculateSHA512(FileInfo file)
+        {
+            using FileStream fileReadStream = file.OpenRead();
+            using System.Security.Cryptography.SHA512 sha = System.Security.Cryptography.SHA512.Create();
+            byte[] hashValueBytes = sha.ComputeHash(fileReadStream);
+            return BitConverter.ToString(hashValueBytes).Replace("-", ""); ;
+        }
+
+        private class ManifestFileVersion
+        {
+            [JsonPropertyOrder(1)]
+            public string version { get; set; }
+        }
+
+        private class ManifestDataV1 : ManifestFileVersion
+        {
+            [JsonPropertyOrder(2)]
```

Comment:
I added the ordering because otherwise, the json was serialized to entries then version
```json
{
  "entries": [
  ...
  ],
  "version": "1"
}
```
so I ordered them just to be consistent, but if parsing can handle it then we dont need the attribute

Reply:
Just checked in a local testing that deserializing still works even if the order is "entries" -> "version"

Reply:
Removed the ordering